Headlines/src/Categories/UIImageEffects.swift

```diff
@@ -0,0 +1,281 @@
+                
+                vImageBoxConvolve_ARGB8888(&effectInBuffer, &effectOutBuffer, nil, 0, 0, radius, radius, nil, imageEdgeExtendFlags)
+                vImageBoxConvolve_ARGB8888(&effectOutBuffer, &effectInBuffer, nil, 0, 0, radius, radius, nil, imageEdgeExtendFlags)
+                vImageBoxConvolve_ARGB8888(&effectInBuffer, &effectOutBuffer, nil, 0, 0, radius, radius, nil, imageEdgeExtendFlags)
+            }
+            
+            var effectImageBuffersAreSwapped = false
+            
+            if hasSaturationChange {
+                let s: CGFloat = saturationDeltaFactor
+                let floatingPointSaturationMatrix: [CGFloat] = [
+                    0.0722 + 0.9278 * s,  0.0722 - 0.0722 * s,  0.0722 - 0.0722 * s,  0,
+                    0.7152 - 0.7152 * s,  0.7152 + 0.2848 * s,  0.7152 - 0.7152 * s,  0,
+                    0.2126 - 0.2126 * s,  0.2126 - 0.2126 * s,  0.2126 + 0.7873 * s,  0,
+                    0,                    0,                    0,  1
+                ]
+                
+                let divisor: CGFloat = 256
+                let matrixSize = floatingPointSaturationMatrix.count
+                var saturationMatrix = [Int16](repeating: 0, count: matrixSize)
+                
+                for i: Int in 0 ..< matrixSize {
+                    saturationMatrix[i] = Int16(round(floatingPointSaturationMatrix[i] * divisor))
+                }
+                
+                if hasBlur {
+                    vImageMatrixMultiply_ARGB8888(&effectOutBuffer, &effectInBuffer, saturationMatrix, Int32(divisor), nil, nil, vImage_Flags(kvImageNoFlags))
+                    effectImageBuffersAreSwapped = true
+                } else {
+                    vImageMatrixMultiply_ARGB8888(&effectInBuffer, &effectOutBuffer, saturationMatrix, Int32(divisor), nil, nil, vImage_Flags(kvImageNoFlags))
+                }
+            }
+            
+            if !effectImageBuffersAreSwapped {
+                effectImage = UIGraphicsGetImageFromCurrentImageContext()!
+            }
+            
+            UIGraphicsEndImageContext()
+            
+            if effectImageBuffersAreSwapped {
+                effectImage = UIGraphicsGetImageFromCurrentImageContext()!
+            }
+            
+            UIGraphicsEndImageContext()
+        }
+        
+        // Set up output context.
+        UIGraphicsBeginImageContextWithOptions(size, false, screenScale)
+        
+        guard let outputContext = UIGraphicsGetCurrentContext() else { return nil }
+        
+        outputContext.scaleBy(x: 1.0, y: -1.0)
+        outputContext.translateBy(x: 0, y: -size.height)
+        
+        // Draw base image.
+        outputContext.draw(cgImage, in: imageRect)
+        
+        // Draw effect image.
+        if hasBlur {
+            outputContext.saveGState()
+            if let maskCGImage = maskImage?.cgImage {
+                outputContext.clip(to: imageRect, mask: maskCGImage);
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Canillitapp/headlines-iOS/blob/b49c86a9b05bec3bbe17833b5a716a8293c42cf1/Headlines/src/Categories/UIImageEffects.swift#L261"></span>Lines should not have trailing semicolons.</td>
    </tr>
  </tbody>
</table>